src/dbnode/integration/peers_bootstrap_single_node_test.go

```diff
@@ -26,18 +26,119 @@ import (
 	"testing"
 	"time"
 
+	"github.com/m3db/m3/src/cluster/services"
+	"github.com/m3db/m3/src/cluster/shard"
 	"github.com/m3db/m3/src/dbnode/integration/generate"
 	"github.com/m3db/m3/src/dbnode/namespace"
 	"github.com/m3db/m3/src/dbnode/retention"
+	"github.com/m3db/m3/src/dbnode/sharding"
 	"github.com/m3db/m3/src/dbnode/storage/bootstrap/bootstrapper"
+	"github.com/m3db/m3/src/dbnode/storage/bootstrap/bootstrapper/uninitialized"
+	"github.com/m3db/m3/src/dbnode/topology"
+	"github.com/m3db/m3/src/dbnode/topology/testutil"
 	xtest "github.com/m3db/m3/src/x/test"
 
 	"github.com/stretchr/testify/require"
 )
 
-// TestPeersBootstrapSingleNode makes sure that we can include the peer bootstrapper
-// in a single-node topology without causing a bootstrap failure or infinite hang.
-func TestPeersBootstrapSingleNode(t *testing.T) {
+// TestPeersBootstrapSingleNodeUninitialized makes sure that we can include the peer bootstrapper
+// in a single-node topology of a non-initialized cluster without causing a bootstrap failure or infinite hang.
+func TestPeersBootstrapSingleNodeUninitialized(t *testing.T) {
```

Comment:
Looks like those two tests share most of the code, please extract it into some test helper fn (as a separate PR).